util/containerutil/shell_shared.go

```diff
@@ -285,24 +286,38 @@ func (cco *commmandContextOutput) string() string {
 
 func (sf *shellFrontend) commandContextStrings(args ...string) (string, []string) {
 	allArgs := append(sf.globalCompatibilityArgs, args...)
-
 	return sf.binaryName, allArgs
 }
 
+func (sf *shellFrontend) commandContextOutputWithRetry(ctx context.Context, retries int, timeout time.Duration, args ...string) (*commmandContextOutput, error) {
+	var err error
+	for i := 0; i < retries; i++ {
+		timeoutCtx, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+		output, cmdErr := sf.commandContextOutput(timeoutCtx, args...)
+		if cmdErr == nil {
+			return output, nil
+		}
+		err = cmdErr
```

Comment:
small nit: In this file we already use `multierror.Append(err, cmdErr)`. It might be nice to see all errors... they might be the same, but they might not be.